bindings/java/src/integration/com/apple/foundationdb/DirectoryTest.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * DirectoryTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+
+import com.apple.foundationdb.directory.DirectoryAlreadyExistsException;
+import com.apple.foundationdb.directory.DirectoryLayer;
+import com.apple.foundationdb.directory.DirectorySubspace;
+import com.apple.foundationdb.directory.NoSuchDirectoryException;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+
+/**
+ * Integration tests for directory logic in FDB. This test requires a running FDB instance to work properly; 
+ * if one cannot be detected, all tests will be skipped.
+ */
+public class DirectoryTest {
+    @ClassRule
+    public static final TestRule dbRule = RequiresDatabase.require();
+    private static final FDB fdb = FDB.selectAPIVersion(700);
```

Comment:
Removed it from the public API, moved comparator() to `ByteArrayUtil`

Reply:
uh...yes, sorry. Lost track of which comment I was replying to. Please disregard.

Most of the formatting _should_ be fixed. We don't appear to have a consistent Java style file (like `.clang-format`), which it's probably worth doing if we consistently encounter such problems.

Reply:
The closest we have to a Java style is this `checkstyle` specification: https://github.com/apple/foundationdb/blob/master/bindings/java/fdb-java-style.xml

But yeah, I agree that we should have something better, ideally something integrated with `.clang-format` if we can so that we only have one style checking tool rather than two...